src/node/sharemanager.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "crypto/cryptobox.h"
+#include "crypto/symmkey.h"
+#include "ds/logger.h"
+#include "genesisgen.h"
+#include "ledgersecrets.h"
+#include "networkstate.h"
+#include "secretshare.h"
+#include "tls/25519.h"
+#include "tls/entropy.h"
+
+#include <vector>
+
+namespace ccf
+{
+  struct LedgerSecretWrappingKey
+  {
+  private:
+    static constexpr auto KZ_KEY_SIZE = crypto::GCM_SIZE_KEY;
+
+  public:
+    std::vector<uint8_t> data; // Referred to as "kz" in TR
+
+    LedgerSecretWrappingKey() : data(tls::create_entropy()->random(KZ_KEY_SIZE))
+    {}
+
+    template <typename T>
+    LedgerSecretWrappingKey(const T& split_secret) :
+      data(
+        std::make_move_iterator(split_secret.begin()),
+        std::make_move_iterator(split_secret.begin() + split_secret.size()))
+    {}
+  };
+
+  class ShareManager
+  {
+  private:
+    NetworkState& network;
+
+  public:
+    ShareManager(NetworkState& network_) : network(network_) {}
+
+    void create(Store::Tx& tx)
```

Comment:
Most of this was moved from `nodestate.h` and the `submitRecoveryShare` handler in `memberfrontend.h`.